<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MotorControl</class>
 <widget class="QMainWindow" name="MotorControl">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>826</width>
    <height>615</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MotorControl</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="qConnectionGroup">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>771</width>
      <height>111</height>
     </rect>
    </property>
    <property name="title">
     <string>Connection</string>
    </property>
    <widget class="QLabel" name="qServerIP">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Server IP:</string>
     </property>
    </widget>
    <widget class="QLabel" name="qServerPort">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>70</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Server Port: </string>
     </property>
    </widget>
    <widget class="QLineEdit" name="qServerIpLine">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>30</y>
       <width>111</width>
       <height>22</height>
      </rect>
     </property>
     <property name="inputMask">
      <string>###.###.###.###</string>
     </property>
     <property name="text">
      <string>192.168.100.23</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="qServerPortSpin">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>70</y>
       <width>111</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>50000</number>
     </property>
     <property name="maximum">
      <number>65535</number>
     </property>
     <property name="value">
      <number>52001</number>
     </property>
    </widget>
    <widget class="QPushButton" name="qConnectButton">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>30</y>
       <width>75</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Connect</string>
     </property>
    </widget>
    <widget class="QPushButton" name="qDisconnectButton">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>70</y>
       <width>75</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Disconnect</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="qMotorDataGroup">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>170</y>
      <width>771</width>
      <height>391</height>
     </rect>
    </property>
    <property name="title">
     <string>Motor Data</string>
    </property>
    <widget class="QGroupBox" name="qSpecificModeGroup">
     <property name="geometry">
      <rect>
       <x>410</x>
       <y>30</y>
       <width>351</width>
       <height>181</height>
      </rect>
     </property>
     <property name="title">
      <string>Specific Mode</string>
     </property>
     <widget class="QLabel" name="qTargetVelocity">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Target Velocity</string>
      </property>
     </widget>
     <widget class="QLabel" name="qTargetPosition">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Target Position</string>
      </property>
     </widget>
     <widget class="QLabel" name="qTargetCurrent">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>140</y>
        <width>101</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Target current(mA)</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="qTargetVelocitySpin">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>40</y>
        <width>81</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>5000</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="qTargetCurrentSpin">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>140</y>
        <width>81</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>5000</number>
      </property>
     </widget>
     <widget class="QPushButton" name="qTargetVelocityButton">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>40</y>
        <width>75</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Send</string>
      </property>
     </widget>
     <widget class="QPushButton" name="qTargetPositionButton">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>70</y>
        <width>75</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Send</string>
      </property>
     </widget>
     <widget class="QPushButton" name="qTargetCurrentButton">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>100</y>
        <width>75</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Send</string>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>70</y>
        <width>21</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string> X</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>100</y>
        <width>21</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string> Y</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="qPositionLineEditX">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>70</y>
        <width>81</width>
        <height>22</height>
       </rect>
      </property>
      <property name="placeholderText">
       <string>double</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="qPositionLineEditY">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>100</y>
        <width>81</width>
        <height>22</height>
       </rect>
      </property>
      <property name="placeholderText">
       <string>double</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="qDirectModeGroup">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>30</y>
       <width>351</width>
       <height>171</height>
      </rect>
     </property>
     <property name="title">
      <string>Direct Mode</string>
     </property>
     <widget class="QLabel" name="qMotorIdLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Motor ID</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="qMotorIdSpin">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>30</y>
        <width>81</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>101</number>
      </property>
      <property name="maximum">
       <number>104</number>
      </property>
     </widget>
     <widget class="QPushButton" name="qDirectModeButton">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>30</y>
        <width>75</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Send</string>
      </property>
     </widget>
     <widget class="QLabel" name="qInstructionLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Instruction</string>
      </property>
     </widget>
     <widget class="QLabel" name="qTypeLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>49</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Type</string>
      </property>
     </widget>
     <widget class="QLabel" name="qValueLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>49</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Value</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="qInstructionSpin">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>60</y>
        <width>81</width>
        <height>21</height>
       </rect>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>137</number>
      </property>
      <property name="value">
       <number>0</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="qTypeSpin">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>90</y>
        <width>81</width>
        <height>21</height>
       </rect>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="qValueSpin">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>120</y>
        <width>81</width>
        <height>21</height>
       </rect>
      </property>
      <property name="minimum">
       <number>-5000</number>
      </property>
      <property name="maximum">
       <number>5000</number>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>220</y>
       <width>351</width>
       <height>131</height>
      </rect>
     </property>
     <property name="title">
      <string>Moving Mode</string>
     </property>
     <widget class="QLabel" name="VerticalVx">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Vertical Vx</string>
      </property>
     </widget>
     <widget class="QLabel" name="RotationTheta">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Rotation Theta</string>
      </property>
     </widget>
     <widget class="QLabel" name="HorizontalVy">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Horizontal Vy</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="qVerticalVxSpin">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>30</y>
        <width>81</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>-5000</number>
      </property>
      <property name="maximum">
       <number>5000</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="qHorizontalVySpin">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>60</y>
        <width>81</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>-5000</number>
      </property>
      <property name="maximum">
       <number>5000</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="qRotationThetaSpin">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>90</y>
        <width>81</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>-5000</number>
      </property>
      <property name="maximum">
       <number>5000</number>
      </property>
     </widget>
     <widget class="QPushButton" name="qMovingModeButton">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>30</y>
        <width>75</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Send</string>
      </property>
     </widget>
    </widget>
    <widget class="QPushButton" name="qResetButton">
     <property name="geometry">
      <rect>
       <x>520</x>
       <y>240</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Reset Motor</string>
     </property>
    </widget>
    <widget class="QPushButton" name="qDirectDriveButton">
     <property name="geometry">
      <rect>
       <x>520</x>
       <y>290</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Direct Drive</string>
     </property>
    </widget>
    <widget class="QPushButton" name="qJoyStickButton">
     <property name="geometry">
      <rect>
       <x>520</x>
       <y>340</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>JoyStick</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>826</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
